staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -524,6 +656,21 @@ func (s *sharedIndexInformer) AddEventHandlerWithResyncPeriod(handler ResourceEv
 	}
 }
 
+func (s *sharedIndexInformer) RemoveEventHandler(handler ResourceEventHandler) bool {
+	for i, listener := range s.processor.listeners {
```

Comment:
I'd consider moving the implementation into `sharedProcessor`. I think we're missing the `listenersLock` here.

Reply:
Makes sense, done